testcontainers/testcontainers-go
# feat: use credential helper in docker config, even if auth is empty in .docker/config.json

## What does this PR do?

If when retrieving the docker configuration, when there's no `auth` field on the user's docker config and if the user has configured credential helpers, the code will use the credential helpers to retrieve the auth information for the image repository instead of setting it to empty. 

## Why is it important?
Credential helpers are often used by third party docker repository providers to perform access control on the img repository.
In my case I'm using [GCP](https://cloud.google.com/container-registry/docs/advanced-authentication#gcloud-helper) but this is the case also for aws and azure. This change will extend the possibility of usage of images that are located in those repositories.

## Related issues
Closes #1078 

## How to test this PR
This is a tricky one to test because of the way the credentials are retrieved using these helpers. 
`cpuguy83/dockercfg` replicates the usage the docker CLI command: it calls an executable located in the PATH, which needs to output the credentials to stdout. This is tricky to emulate on go because one would need to use a script to link an executable that would mock the behaviour of a credential helper to the PATH when running the test. 
The other solution would involve having a private registry setup in a private registry provider such as GCP and use the credential helper of the provider. That would also imply some dependencies on the tests. 
Fortunately I had access to a private registry, so I was able to verify it by just using my fork in the project where I was using an image of the priv repository in my tests.
